### ISSUE TITLE
No cut/copy/paste in Configurator (File Explorer) 

### ISSUE THREAD

**Issue by uspino2** [danielperna84/hass-configurator#191]
Ever since the last few updates I lost the ability to copy and paste bits of text code in Configurator (File Explorer now). Shortcuts work, but no copy/cut/paste option in the right-click menu. I'm using Safari.

**Comment by danielperna84** [comment]
I have to admit that I have never payed attention to it since I only use keyboard shortcuts for this. But indeed there are no copy/paste options in the context menu.
That being said, I don't think I can change anything about that. This seems to be a problem with the Ace Editor that is being used. [Here](https://github.com/ajaxorg/ace/search?q=copy&type=Issues) you can see lots of issues regarding copy/paste.

**Comment by Kisty** [comment]
Same for Android. Guessing it's not just Windows. If this is a bug with Ace, has it been reported there?

**Comment by Kisty** [comment]
This seems to be fixed in this example. Would an update of Ace help? https://ace.c9.io/build/kitchen-sink.html

**Comment by danielperna84** [comment]
I have added the latest version (1.4.8 I believe) in the latest configurator. I think this is actually a regression that has happened within Ace, and the example you are pointing to is out of date.

**Comment by Kisty** [comment]
@danielperna84 strange. The example is running 1.4.8 though.

**Comment by danielperna84** [comment]
Ok, then feel free to research why it's not working within the configurator. 😉

**Comment by uspino2** [comment]
Could it have something to do with this?

[](https://www.techiedelight.com/disabling-text-selection-cut-copy-right-click/)

**Comment by Kisty** [comment]
@danielperna84 I've had a good look and I can't see anything obvious that's preventing things, but I think it's to do with the JavaScript code in [`dev.html`](https://github.com/danielperna84/hass-configurator/blob/master/hass_configurator/dev.html). I think it's either overriding text selection or the right click context menu behaviour.

**Comment by danielperna84** [comment]
> Could it have something to do with this?
> 
> [https://www.techiedelight.com/disabling-text-selection-cut-copy-right-click/](url)
> 
> Cut/cupy/paste is not allowed in Chrome either.

No, I don't think that's the problem. I have note implemented anything consciously that would prevent copy / paste. That wouldn't make any sense for a text editor.

@Kisty 
It's either a side-effect of something in `dev.html` that's not directly related, or its a result of not using 100% of the Ace Editor code. Or maybe their build on cdjnjs is broken (the demo you refer to does not make use of the hosted version).

**Comment by erkr** [comment]
6 years passed. Currently all other editors in the front end support copy paste. Copy and paste on a mobile is really a pain in🫣

Hitting the 3 dots without losing the selection succeeds one out of 3 times for me, and there is only a copy, no paste!!
![image](https://github.com/user-attachments/assets/dffbe3d9-0894-4941-bbe1-a95f6a070547)
